prime-angular-frontend/src/app/modules/enrolment/pages/care-setting/care-setting.component.ts

```diff
@@ -233,16 +221,20 @@ export class CareSettingComponent extends BaseEnrolmentProfilePage implements On
     super.nextRouteAfterSubmit(nextRoutePath);
   }
 
-  private removeIncompleteCareSettings() {
-    this.careSettings.controls
-      .forEach((control: FormGroup, index: number) => {
-        const value = control.get('careSettingCode').value;
+  private removeIncompleteCareSettings(allowEmptyCareSettings: boolean = false) {
+    this.enrolmentFormStateService.json.careSettings.forEach((careSetting: CareSetting) => {
```

Comment:
You've got a solution, but does it make for clean code?  You're looping over a list and then within the loop using a method that loops over a list to find an index to be removed based on a predicate.

Reply:
You're using the JSON, but remember when we were talking in Slack I indicated you just need `this.careSetting.value` not invoking the `.json` getter.

Reply:
Also, I mentioned that you would use `forEach` but more needs to happen before you use it.  Also, that you were on the right track with filter, but didn't take it far enough.

Reply:
Refactored as discussed, made minor changes to cover certain cases